src/module.d.ts

```diff
@@ -0,0 +1,401 @@
+
+export interface SettingsProps {
+    settingsComponents?: GriddleComponent<any>[];
+    className?: string;
+    style?: React.CSSProperties;
+}
+
+class Settings extends React.Component<SettingsProps, any> {
+}
+
+const SettingsContainer: (OriginalComponent: any) => any;
+
+const SettingsComponents: PropertyBag<GriddleComponent<any>>;
+
+} // namespace components
+
+export interface GriddleComponents {
+    Cell?: any;
+    CellContainer?: any;
+    ColumnDefinition?: any;
+    Row?: any;
+    RowContainer?: any;
+    RowDefinition?: any;
+    Table?: any;
+    TableBody?: any;
+    TableBodyContainer?: any;
+    TableHeading?: any;
+    TableHeadingContainer?: any;
+    TableHeadingCell?: any;
+    TableHeadingCellContainer?: any;
+    TableHeadingCellEnhancer?: any;
+    TableContainer?: any;
+    Layout?: any;
+    LayoutContainer?: any;
+    NextButton?: any;
+    NextButtonEnhancer?: any;
+    NextButtonContainer?: any;
+    NoResults?: any;
+    NoResultsContainer?: any;
+    PageDropdown?: any;
+    PageDropdownContainer?: any;
+    Pagination?: any;
+    PaginationContainer?: any;
+    PreviousButton?: any;
+    PreviousButtonEnhancer?: any;
+    PreviousButtonContainer?: any;
+    Filter?: any;
+    FilterEnhancer?: any;
+    FilterContainer?: any;
+    SettingsToggle?: any;
+    SettingsToggleContainer?: any;
+    SettingsWrapper?: any;
+    SettingsWrapperContainer?: any;
+    Settings?: any;
+    SettingsContainer?: any;
+    SettingsComponents?: PropertyBag<GriddleComponent<any>>;
+}
+
+export interface GriddleEvents {
+    onFilter?: (filterText: string) => void;
+    onSort?: (sortProperties: any) => void;
+    onNext?: () => void;
+    onPrevious?: () => void;
+    onGetPage?: (pageNumber: number) => void;
+}
+
+export interface GriddleSortKey {
+    id: string;
+    sortAscending: boolean;
+}
+
+export interface GriddleStyleElements<T> {
+    Cell?: T;
+    Filter?: T;
+    Loading?: T;
+    NextButton?: T;
+    NoResults?: T;
+    PageDropdown?: T;
+    Pagination?: T;
+    PreviousButton?: T;
+    Row?: T;
+    RowDefinition?: T;
+    Settings?: T;
+    SettingsToggle?: T;
+    Table?: T;
+    TableBody?: T;
+    TableHeading?: T;
+    TableHeadingCell?: T;
+    TableHeadingCellAscending?: T;
+    TableHeadingCellDescending?: T;
+}
+
+export interface GriddleStyleIcons {
+    sortAscendingIcon?: any;
+    sortDescendingIcon?: any;
+}
+
+export interface GriddleStyleConfig {
+    classNames?: GriddleStyleElements<string>;
+    icons?: GriddleStyleElements<GriddleStyleIcons>;
+    styles?: GriddleStyleElements<React.CSSProperties>;
+}
+
+export interface GriddlePageProperties {
+    currentPage?: number;
+    pageSize?: number;
+    recordCount?: number;
+}
+
+interface RowRenderProperties {
+    rowKey?: string;
+    childColumnName?: string;
+    props?: {
+        children: components.ColumnDefinition[];
+    };
+}
+
+interface ColumnRenderProperties {
+    id: string;
```

Comment:
One question I just stumbled on: is `id` actually required here on `columnProperties` entries? I found [this fallback](https://github.com/GriddleGriddle/Griddle/blob/ecc8ee21aed15b2507b9b1fd812cff8a16190fcf/src/selectors/dataSelectors.js#L225) to the `columnProperties` key if `id` is missing, but I can't tell if that's meant to be a supported usage.